### ISSUE TITLE
🐛 Inconsistent monorepo lint results

### ISSUE THREAD

**Issue by ben-xiao-io** [biomejs/biome-vscode#768]
### VS Code version

1.99.3

### Extension version

3.2.0

### Biome version

2.2.2

### Operating system

- [ ] Windows
- [x] macOS
- [ ] Linux

### Description

There are two `biome.json` files, one in the root and one in `apps./project-bad`. The latter extends from the one in the root. However, if I remove the `root: false` out of the child one, the results are very inconsistent. From my understanding, the `root` key is optional ([Docs](https://next.biomejs.dev/guides/big-projects/))

Reproduce repo: https://github.com/ben-xiao-io/biome-parser-reproduction

Here is a screen recording:
https://github.com/user-attachments/assets/73df50c8-b4e3-4bc8-b13a-68cba947b0ba

Note that using biome cli directly does work. `bunx biome lint` returns no errors everytie.

### Steps to reproduce

1. Clone reproduction repo, and `bun i` -- bun (v1.2.19)
2. `Cmd + Shift + P` --> `Biome: Restart` a bunch of times

### Expected behavior

Should not produce any errors every time.

### Does this issue occur when using the CLI directly?

No

### Link to a minimal reproduction

https://github.com/ben-xiao-io/biome-parser-reproduction

### Logs

```plain text

```

**Comment by ematipico** [comment]
Can you please share your VSCode settings?

**Comment by ben-xiao-io** [comment]
Updated the repo. adding it here for vis

```
{
  "python.testing.unittestArgs": ["-v", "-s", "-p", "*test.py"],
  "python.testing.pytestEnabled": false,
  "python.testing.unittestEnabled": true,
  // Format on save globally
  "editor.formatOnSave": true,
  // Prettier by default
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "biome.requireConfiguration": false,
  // Force Biome for JS
  "[javascript][javascriptreact][typescript][typescriptreact][jsonc][css]": {
    "editor.defaultFormatter": "biomejs.biome",
    "editor.tabSize": 2
  },
  "[python]": {
    "editor.formatOnSave": true,
    "editor.tabSize": 4,
    "editor.defaultFormatter": "charliermarsh.ruff"
  },
  "[prisma]": {
    "editor.defaultFormatter": "Prisma.prisma"
  },

  "editor.codeActionsOnSave": {
    // Biome
    "source.fixAll.biome": "explicit",
    "source.organizeImports.biome": "explicit",
    // Ruff
    "source.fixAll.ruff": "explicit",
    "source.organizeImports.ruff": "explicit"
  }
}
```
